src/turbomind/layers/sampling_layers/TopPSamplingLayer.cu

```diff
@@ -198,60 +137,52 @@ void TopPSamplingLayer<T>::setup(const size_t batch_size, const size_t beam_widt
     **/
 
     TM_LOG_DEBUG(__PRETTY_FUNCTION__);
-    BaseSamplingLayer<T>::setup(batch_size, beam_width, runtime_args);
     const Tensor runtime_top_p = runtime_args->isExist("runtime_top_p") ? runtime_args->at("runtime_top_p") : Tensor();
     const size_t runtime_top_p_size = runtime_top_p.size();
-    if (runtime_top_p_size == 0) {
-        std::fill_n(skip_decode_, batch_size, true);
+    const Tensor runtime_top_k = runtime_args->isExist("runtime_top_k") ? runtime_args->at("runtime_top_k") : Tensor();
+    const size_t runtime_top_k_size = runtime_top_k.size();
+    uint         min_top_k          = runtime_top_k_size > 0 ? runtime_top_k.min<uint>() : 0;
+    skip_all_                       = false;
+
+    if (runtime_top_p_size == 0 || min_top_k > 0) {
```

Comment:
@lzhangzz @irexyc @lvhan028 Do you have any suggestions?

Reply:
The topk and topk sampling layers are fused layers. Either one of these two layers will do topk and topk sampling. 

If top_k > 0, chose topk sampling layer, otherwise chose topp sampling layer.

Reply:
ok